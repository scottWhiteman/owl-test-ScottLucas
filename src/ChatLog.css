#ChatLog {
    background-color: #fff;
    border: 3px solid black;
}