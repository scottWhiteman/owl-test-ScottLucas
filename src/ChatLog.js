import React from 'react';
import './ChatLog.css';
import store from './store';

class ChatLog extends React.Component {
  generateLog() {
    return store.chatEvents.map(chatEvent => {
      let currentUser = store.participants.find(participant => chatEvent.participantId === participant.id).name;

      console.log(currentUser);

      if (chatEvent.type === 'message') {
        let time = new Date(chatEvent.timestamp);
        let timeString = `${time.getHours().toString()}:${time.getMinutes().toString()}`;

        return (
          <div>
            <p>{currentUser} {timeString}</p>
            <p>{chatEvent.message}</p>
          </div>
        );
      } else {
        return (
          <div>
            <p>{currentUser} gave a {chatEvent.type}.</p>
          </div>
        );
      }
    })
  }
  
  render() {
    const elems = this.generateLog();
    console.log(elems);
    
    return (
      <div id="ChatLog">
        {elems}
      </div>
    );
  }
}

export default ChatLog;