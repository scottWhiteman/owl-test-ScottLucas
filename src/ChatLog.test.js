import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ChatLog from './ChatLog';

it ("Participant List render without crash", () => {
    const div = document.createElement("div");
    ReactDOM.render(
        <ChatLog />
    , div);
    ReactDOM.unmountComponentAtNode(div);
});

it ("Participants render without crash", () => {
    const tree = renderer.create(<ChatLog />).toJSON();
    expect(tree).toMatchSnapshot();
});
