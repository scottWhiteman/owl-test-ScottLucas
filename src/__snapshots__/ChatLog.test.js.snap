// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Participants render without crash 1`] = `
<div>
  <div>
    <p>
      Axe Kubicka
       
      4:48
    </p>
    <p>
      Hello world
    </p>
  </div>
  <div>
    <p>
      Caty Flucker
       gave a 
      thumbs-up
      .
    </p>
  </div>
  <div>
    <p>
      Frank Runciman
       gave a 
      thumbs-down
      .
    </p>
  </div>
  <div>
    <p>
      Koren Templeton
       gave a 
      raise-hand
      .
    </p>
  </div>
  <div>
    <p>
      Koren Templeton
       gave a 
      clap
      .
    </p>
  </div>
  <div>
    <p>
      Koren Templeton
       gave a 
      join
      .
    </p>
  </div>
  <div>
    <p>
      Ashla Attwool
       gave a 
      leave
      .
    </p>
  </div>
  <div>
    <p>
      Axe Kubicka
       gave a 
      join-stage
      .
    </p>
  </div>
  <div>
    <p>
      Axe Kubicka
       gave a 
      leave-stage
      .
    </p>
  </div>
</div>
`;
